streamSignupManager/streamSignupManager.py

```diff
@@ -0,0 +1,1089 @@
+import discord
+import asyncio
+from redbot.core import Config
+from redbot.core import commands
+from redbot.core import checks
+from redbot.core.utils.predicates import ReactionPredicate
+from redbot.core.utils.menus import start_adding_reactions
+
+defaults = {"Applications": {}, "Schedule": {}, "TimeSlots": {}, "LiveStreamChannels": [], "StreamFeedChannel": None}
+verify_timeout = 30
+
+# TODO: 
+# - make [p]clearStreamSchedule notify players of stream removal
+# + (maybe?) reject applications for same time frame/alert that a different application has been accepted.
+
+# Roles: Captain, GM, <Tier>, <Franchise>, Media Committee
+
+class StreamSignupManager(commands.Cog):
+    def __init__(self, bot):
+        self.config = Config.get_conf(self, identifier=1234567892, force_registration=True)
+        self.config.register_guild(**defaults)
+        self.match_cog = bot.get_cog("Match")
+        self.team_manager_cog = bot.get_cog("TeamManager")
+        self.bot = bot
+        self.MEDIA_ROLE_STR = "Media Committee"
+
+        # Application statuses
+        self.PENDING_OPP_CONFIRMATION_STATUS = "PENDING_OPP_CONFIRMATION"
+        self.PENDING_LEAGUE_APPROVAL_STATUS = "PENDING_LEAGUE_APPROVAL"
+        self.SCHEDULED_ON_STREAM_STATUS = "SCHEDULED_ON_STREAM"
+        self.REJECTED_STATUS = "REJECTED"
+        self.RESCINDED_STATUS = "RESCINDED"
+    
+    @commands.command(aliases=["streamapp", "streamapply", "streamApplications", "streamapplications"])
+    @commands.guild_only()
+    async def streamApp(self, ctx, action, match_day, time_slot=None, team=None):
```

Comment:
For the action parameter I think a lot of the code around checking which action it is can be streamlined by using subcommands. That is a built in way to allow multi-word commands in order to be more specific with what the user is trying to do with the command. Look at the sixMans cog, specifically the queueLeaderBoard command and its subcommands such as overall, day, week, etc. Let me know if you have any questions about that.

Reply:
Done.